static/app/views/routeNotFound.tsx

```diff
@@ -1,44 +1,46 @@
-import {Component} from 'react';
+import {useEffect} from 'react';
 import DocumentTitle from 'react-document-title';
+import {RouteComponentProps} from 'react-router';
 import * as Sentry from '@sentry/react';
-import {Location} from 'history';
 
 import NotFound from 'app/components/errors/notFound';
 import Footer from 'app/components/footer';
 import Sidebar from 'app/components/sidebar';
+import {t} from 'app/locale';
 
-type Props = {
-  location: Location;
-};
+type Props = RouteComponentProps<{}, {}>;
+
+function RouteNotFound({router, location}: Props) {
+  useEffect(() => {
+    // Attempt to fix trailing slashes first
+    const {pathname, search, hash} = location;
+
+    if (pathname[pathname.length - 1] !== '/') {
+      router.replace(`${pathname}/${search}${hash}`);
+      return;
+    }
 
-class RouteNotFound extends Component<Props> {
-  componentDidMount() {
     Sentry.withScope(scope => {
       scope.setFingerprint(['RouteNotFound']);
       Sentry.captureException(new Error('Route not found'));
     });
-  }
-
-  getTitle = () => 'Page Not Found';
+  }, [location.pathname]);
```

Comment:
```suggestion
  }, []);
```

Would give you ~= didMount

Reply:
I left it here incase we have a link on the 404 page that for whatever reason has a missing `/`, in which case the component would render with updated props, but not 'on mount'